### PR TITLE
CAL-119 Restored image chipping action

### PR DESCRIPTION
**Pull request by lcrosenbu**
#### What does this PR do?
Fixed image chipping action provider that was recently broken. This PR restores the generation of the original image, which is required for the image chipping function. Additionally, this removes the extra `Export as overview-chip` option from the Actions tab.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@dcruver 
@bdeining 
@emmberk 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jlcsmith
@coyotesqrl 

#### How should this be tested?
1. Build/run alliance.
2. Ingest Image NITF.
3. Perform search to retrieve ingested NITF.
4. Verify the "Export as overview-chip" option is no longer available. 
5. Verify "Chip Image" function is present under Actions and works.
6. Additionally, ingest GMTI NITF to ensure "Chip Image" is not available in Actions list.

#### Any background context you want to provide?
Dependent on PR https://github.com/codice/ddf/pull/1173.

#### What are the relevant tickets?
https://codice.atlassian.net/browse/CAL-119

#### Screenshots (if appropriate)
N/A 

#### Checklist:
- [ ] Documentation Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by bdeining** [comment]
LGTM After logging change

File: catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java [hunk 1]
```diff
@@ -78,12 +79,19 @@ public String getId() {
     public <T> boolean canHandle(T subject) {
         if (subject instanceof Metacard) {
             Metacard metacard = (Metacard) subject;
-            String contentTypeName = metacard.getContentTypeName();
-            if (NITF_CONTENT_TYPE.equals(contentTypeName) && metacard.getLocation() != null) {
-                return true;
+
+            Attribute attribute = metacard.getAttribute(Media.TYPE);
```
Comment (jlcsmith) [inline catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: It seems to me that the canHandle() method should be checking if the metacard is an image NITF, has a location, and has the expected derived images.

**Comment by emmberk** [comment]
Successfully hero'd. View original, View Overview, and Chip Image all work as expected. The .ntf file, the original, and the overview can be found in data/content/store.

**Comment by dcruver** [comment]
👍

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java [hunk 2]
```diff
@@ -223,4 +247,53 @@ private String buildOverviewTitle(String title) {
...
+
+        int newWidth = (int) Math.round(width * (maxSideLength / height));
+        return newWidth;
+    }
+
+    public void setMaxSideLength(int maxSideLength) {
+        if (maxSideLength > 0) {
+            this.maxSideLength = maxSideLength;
+        } else {
+            LOGGER.warn(
```
Comment (jlcsmith) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java]: debug level please

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java [hunk 3]
```diff
@@ -223,4 +247,53 @@ private String buildOverviewTitle(String title) {
...
+                    "method argument 'maxSideLength' must be greater than 0.  Using default value instead.");
+            return;
+        }
+
+        LOGGER.debug(String.format("Setting maxSideLength to: %s", maxSideLength));
```
Comment (lcrosenbu) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java]: Fixed.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java (hunk 2 above)
Comment (lcrosenbu) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java]: Keeping this log level at `warn` since this is a startup issue only where the blueprint config would need to be updated with a valid value.

File: catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java [hunk 4]
```diff
@@ -76,14 +81,49 @@ public String getId() {
...
-                return true;
+
+            boolean isImageNitf = IMAGE_NITF_METACARD_TYPE
+                    .equals(metacard.getMetacardType().getName());
+            boolean hasLocation = metacard.getLocation() != null;
+            boolean hasDerivedImage = hasOriginalDerivedResource(metacard); // original is required
+                                                                            // for chipping
+
+            canHandle = isImageNitf && hasLocation && hasDerivedImage;
+            LOGGER.debug(
```
Comment (jrnorth) [inline catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: Seems like this should be `trace`.

File: catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java [hunk 5]
```diff
@@ -76,14 +81,49 @@ public String getId() {
...
+            for (Serializable attributeValue : attribute.getValues()) {
+                Serializable serializable = attributeValue;
+                if (serializable instanceof String) {
+                    String uriString = (String) serializable;
+                    try {
+                        URI derivedResourceUri = new URI(uriString);
+
+                        // find the #original URI fragment
+                        if (ORIGINAL_QUALIFIER.equals(derivedResourceUri.getFragment())) {
+                            LOGGER.debug("Found the {} URI fragment in {}",
```
Comment (jrnorth) [inline catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: Should this be `trace`?

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/MetacardFactory.java [hunk 6]
```diff
@@ -52,6 +55,9 @@ public Metacard createMetacard(String id) {
         metacard.setAttribute(new AttributeImpl(Media.TYPE, MIME_TYPE.toString()));
         metacard.setAttribute(new AttributeImpl(Metacard.CONTENT_TYPE, MIME_TYPE.toString()));
 
+        LOGGER.debug("Setting the metacard attribute [{}, {}], [{}, {}]", Media.TYPE, MIME_TYPE,
```
Comment (jrnorth) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/MetacardFactory.java]: Probably `trace`.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java [hunk 7]
```diff
@@ -111,48 +115,53 @@ private boolean isNitfMimeType(String rawMimeType) {
...
+
+                contentItems.add(originalImageContentItem);
             }
-        } catch (IOException | ParseException | NitfFormatException | UnsupportedOperationException e) {
+        } catch (IOException | ParseException | NitfFormatException e) {
             LOGGER.warn(e.getMessage(), e);
         }
 
-        return Optional.empty();
+        return;
```
Comment (jrnorth) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java]: Don't need this for a void method.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java [hunk 8]
```diff
@@ -223,4 +225,52 @@ private String buildOverviewTitle(String title) {
...
+
+        metacard.setAttribute(attribute);
+    }
+
+    private int calculateOverviewHeight(BufferedImage image) {
+        final int width = image.getWidth();
+        final int height = image.getHeight();
+
+        if (width >= height) {
+            int newHeight = (int) Math.round(((height * (maxSideLength / width))));
```
Comment (jrnorth) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java]: Extra parentheses.

Comment (lcrosenbu) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java]: The parens are correct.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java (hunk 7 above)
Comment (lcrosenbu) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java]: Removed.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java (hunk 8 above)
Comment (jrnorth) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java]: `Math.round(height * (maxSideLength / width))`? There are two extra pairs.

File: catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java [hunk 9]
```diff
@@ -76,14 +81,43 @@ public String getId() {
...
+        }
+
+        return canHandle;
+    }
+
+    private boolean hasOriginalDerivedResource(Metacard metacard) {
+        Attribute attribute = metacard.getAttribute(Core.DERIVED_RESOURCE_URI);
+        if (attribute != null) {
+            for (Serializable attributeValue : attribute.getValues()) {
+                Serializable serializable = attributeValue;
```
Comment (jrnorth) [inline catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: Don't need this extra variable.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java [hunk 10]
```diff
@@ -223,4 +223,52 @@ private String buildOverviewTitle(String title) {
...
+
+        metacard.setAttribute(attribute);
+    }
+
+    private int calculateOverviewHeight(BufferedImage image) {
+        final int width = image.getWidth();
+        final int height = image.getHeight();
+
+        if (width >= height) {
+            int newHeight = (int) Math.round(height * (maxSideLength / width));
```
Comment (jrnorth) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java]: Could return the result directly without a local variable.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java [hunk 11]
```diff
@@ -223,4 +223,52 @@ private String buildOverviewTitle(String title) {
...
+
+    private int calculateOverviewWidth(BufferedImage image) {
+        final int width = image.getWidth();
+        final int height = image.getHeight();
+
+        if (width >= height) {
+            return Math.min(width, (int) maxSideLength);
+        }
+
+        int newWidth = (int) Math.round(width * (maxSideLength / height));
```
Comment (jrnorth) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java]: Same here.

**Comment by jrnorth** [comment]
LGTM

**Comment by jlcsmith** [comment]
Built and tested - LGTM 👍

File: catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java [hunk 12]
```diff
@@ -56,9 +62,8 @@
             URL url;
             String chipPath = null;
             try {
-                chipPath =
-                        SystemBaseUrl.getBaseUrl() + PATH + "?id=" + metacard.getId() + "&source="
-                                + metacard.getSourceId();
+                chipPath = SystemBaseUrl.getBaseUrl() + PATH + "?id=" + metacard.getId()
```
Comment (coyotesqrl) [inline catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: Should probably use `String.format()` instead of all those concats.

File: catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java [hunk 13]
```diff
@@ -76,14 +81,42 @@ public String getId() {
...
+            boolean hasLocation = metacard.getLocation() != null;
+            boolean hasDerivedImage = hasOriginalDerivedResource(metacard); // original is required for chipping
+
+            canHandle = isImageNitf && hasLocation && hasDerivedImage;
+        }
+
+        return canHandle;
+    }
+
+    private boolean hasOriginalDerivedResource(Metacard metacard) {
```
Comment (coyotesqrl) [inline catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: This method has high cyclomatic complexity. You might consider this functional approach:
```java
    private boolean hasOriginalDerivedResource(Metacard metacard) {
        Attribute attribute = metacard.getAttribute(Core.DERIVED_RESOURCE_URI);

        return Stream.of(attribute)
                .filter(Objects::nonNull)
                .flatMap(a -> a.getValues()
                        .stream())
                .filter(String.class::isInstance)
                .map(String.class::cast)
                .anyMatch(this::hasOriginalQual);
    }

    private boolean hasOriginalQual(String uriString) {
        try {
            URI derivedResourceUri = new URI(uriString);

            // find the #original URI fragment
            if (ORIGINAL_QUALIFIER.equals(derivedResourceUri.getFragment())) {
                return true;
            }
        } catch (URISyntaxException use) {
            LOGGER.debug("Could not parse URI string [{}]", uriString);
        }
        return false;
    }
```

File: catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java [hunk 14]
```diff
@@ -26,22 +29,25 @@
...
-    public static final String DESCRIPTION =
-            "Opens a new window to enter the boundaries of an image chip for a Metacard.";
+    public static final String DESCRIPTION = "Opens a new window to enter the boundaries of an image chip for a Metacard.";
 
     public static final String PATH = "/chipping/chipping.html";
 
     public static final String ID = "catalog.data.metacard.image.chipping";
 
-    public static final String NITF_CONTENT_TYPE = "image/nitf";
+    public static final String IMAGE_NITF_METACARD_TYPE = "image";
```
Comment (coyotesqrl) [inline catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: This seems like a pretty generic name for a nitf metacard type. Are we concerned we might bump up against other image types in the future - created by us or by integrators - where we might mistakenly think we can handle a metacard that doesn't contain a nitf?

File: catalog/imaging/imaging-actionprovider-chip/src/test/java/org/codice/alliance/imaging/chip/actionprovider/TestImageChipActionProvider.java [hunk 15]
```diff
@@ -64,8 +71,8 @@ public void testCanHandleInvalidMetacard() {
 
     @Test
     public void testCanHandleInvalidLocationOnMetacard() {
```
Comment (coyotesqrl) [inline catalog/imaging/imaging-actionprovider-chip/src/test/java/org/codice/alliance/imaging/chip/actionprovider/TestImageChipActionProvider.java]: Nit: the name of this test implies to me that you think it "can handle and invalid location". I'd change to something like `testDoesNotHandleInvalidLocation()`

Comment (coyotesqrl) [inline catalog/imaging/imaging-actionprovider-chip/src/test/java/org/codice/alliance/imaging/chip/actionprovider/TestImageChipActionProvider.java]: Should also add a negative test for when the location isn't null but is an invalid location string.

File: catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java (hunk 12 above)
Comment (coyotesqrl) [inline catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: Outside the scope of your changes, but the return value of this method should probably be changed to 
`return Collections.singletonList(new ActionImpl(getId(), TITLE, DESCRIPTION, url));`

File: catalog/imaging/imaging-actionprovider-chip/src/test/java/org/codice/alliance/imaging/chip/actionprovider/TestImageChipActionProvider.java [hunk 16]
```diff
@@ -98,21 +105,21 @@ public void testGetActionsNullMetacard() {
...
+        imageMetacard.setLocation(null);
+        assertThat(imagingChipActionProvider.getActions(imageMetacard), hasSize(0));
     }
 
     @Test
     public void testGetActionsUnsupportedProtocolOnMetacard() {
         String protocol = SystemBaseUrl.getProtocol();
         System.setProperty("org.codice.ddf.system.protocol", "udp://");
-        assertThat(imagingChipActionProvider.getActions(metacard), hasSize(0));
+        assertThat(imagingChipActionProvider.getActions(imageMetacard), hasSize(0));
```
Comment (coyotesqrl) [inline catalog/imaging/imaging-actionprovider-chip/src/test/java/org/codice/alliance/imaging/chip/actionprovider/TestImageChipActionProvider.java]: I'm unclear why changing the protocol does this. Nothing that I could see in the implementation of the action provider should care about this and it's no javadocs indicating that the protocol has an effect.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/MetacardFactory.java [hunk 17]
```diff
@@ -27,21 +31,20 @@
 
     private static final String ID = "ddf/catalog/transformer/nitf";
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(NitfPreStoragePlugin.class);
```
Comment (coyotesqrl) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/MetacardFactory.java]: Wrong class passed to the logger.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/SegmentHandler.java [hunk 18]
```diff
@@ -44,7 +44,7 @@
         AttributeDescriptor descriptor = attribute.getAttributeDescriptor();
 
         if (descriptor == null) {
-            LOGGER.error("Could not set metacard attribute " + attribute.getLongName()
+            LOGGER.warn("Could not set metacard attribute " + attribute.getLongName()
```
Comment (coyotesqrl) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/SegmentHandler.java]: Use param interpolation.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java [hunk 19]
```diff
@@ -111,48 +115,51 @@ private boolean isNitfMimeType(String rawMimeType) {
 
     private void process(List<ContentItem> contentItems) {
         List<ContentItem> newContentItems = new LinkedList<>();
```
Comment (coyotesqrl) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java]: Nit: it's pretty unusual to use a java `LinkedList` and not an `ArrayList`. There are cases for it, of course. Is there a performance benefit expected from using it here?

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java [hunk 20]
```diff
@@ -223,4 +223,51 @@ private String buildOverviewTitle(String title) {
...
+
+        return (int) Math.round(width * (maxSideLength / height));
+    }
+
+    public void setMaxSideLength(int maxSideLength) {
+        if (maxSideLength > 0) {
+            LOGGER.debug("Setting derived image maxSideLength to {}", maxSideLength);
+            this.maxSideLength = maxSideLength;
+        } else {
+            LOGGER.warn(
```
Comment (coyotesqrl) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java]: An `info` or `debug` level log is probably sufficient here. There's nothing for the admin to fix, necessarily, as everything will work just fine.

File: catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java (hunk 12 above)
Comment (lcrosenbu) [inline catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: Done.

File: catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java (hunk 13 above)
Comment (lcrosenbu) [inline catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: Refactored with suggestion.

File: catalog/imaging/imaging-actionprovider-chip/src/test/java/org/codice/alliance/imaging/chip/actionprovider/TestImageChipActionProvider.java (hunk 15 above)
Comment (lcrosenbu) [inline catalog/imaging/imaging-actionprovider-chip/src/test/java/org/codice/alliance/imaging/chip/actionprovider/TestImageChipActionProvider.java]: Renamed test methods for clarity.

Comment (lcrosenbu) [inline catalog/imaging/imaging-actionprovider-chip/src/test/java/org/codice/alliance/imaging/chip/actionprovider/TestImageChipActionProvider.java]: Bug discovered! Updated code to perform more robust validation checking on location format, beyond simple non-null check. Added unit test for non-null, invalid WKT location.

File: catalog/imaging/imaging-actionprovider-chip/src/test/java/org/codice/alliance/imaging/chip/actionprovider/TestImageChipActionProvider.java (hunk 16 above)
Comment (lcrosenbu) [inline catalog/imaging/imaging-actionprovider-chip/src/test/java/org/codice/alliance/imaging/chip/actionprovider/TestImageChipActionProvider.java]: Removed this obsolete unit test.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/MetacardFactory.java (hunk 17 above)
Comment (lcrosenbu) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/MetacardFactory.java]: Oops, fixed.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/SegmentHandler.java (hunk 18 above)
Comment (lcrosenbu) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/SegmentHandler.java]: Done.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java (hunk 19 above)
Comment (lcrosenbu) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java]: Refactored to use `ArrayList`.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java (hunk 20 above)
Comment (lcrosenbu) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/image/NitfPreStoragePlugin.java]: This is the third time revisiting this, so I modified the log level to `info`.

File: catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java [hunk 21]
```diff
@@ -56,16 +70,20 @@
             URL url;
             String chipPath = null;
             try {
-                chipPath =
-                        SystemBaseUrl.getBaseUrl() + PATH + "?id=" + metacard.getId() + "&source="
-                                + metacard.getSourceId();
+                chipPath = String.format("%1s%2s?id=%3s&source=%4s",
```
Comment (coyotesqrl) [inline catalog/imaging/imaging-actionprovider-chip/src/main/java/org/codice/alliance/imaging/chip/actionprovider/ImagingChipActionProvider.java]: This is fine, but for future reference you don't need to specify the position of the tokens. `%s%s?id=%s&source=%s` would work fine.

**Comment by coyotesqrl** [comment]
👍
